bitnami/zookeeper/templates/_helpers.tpl

```diff
@@ -41,40 +52,92 @@ Check if there are rolling tags in the images
 {{- end -}}
 
 {{/*
-Return ZooKeeper Client Password
+Return the ZooKeeper authentication credentials secret
 */}}
-{{- define "zookeeper.clientPassword" -}}
-{{- if .Values.auth.clientPassword -}}
+{{- define "zookeeper.secretName" -}}
+{{- if .Values.auth.existingSecret -}}
+    {{- printf "%s" (tpl .Values.auth.existingSecret $) -}}
+{{- else -}}
+    {{- printf "%s-auth" (include "common.names.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a ZooKeeper authentication credentials secret object should be created
+*/}}
+{{- define "zookeeper.createSecret" -}}
+{{- if and .Values.auth.enabled (empty .Values.auth.existingSecret) -}}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the available value for certain key in an existing secret (if it exists),
+otherwise it generates a random value.
+*/}}
+{{- define "getValueFromSecret" }}
+    {{- $len := (default 16 .Length) | int -}}
+    {{- $obj := (lookup "v1" "Secret" .Namespace .Name).data -}}
+    {{- if $obj }}
+        {{- index $obj .Key | b64dec -}}
+    {{- else -}}
+        {{- randAlphaNum $len -}}
+    {{- end -}}
+{{- end }}
```

Comment:
I have seen this function replicated in several chart standardization, IMHO we should add it to the common chart.

Reply:
Totally agree! I'll do that in a separate contribution.

Reply:
👍